lib/Doctrine/DBAL/Types/ConversionException.php

```diff
@@ -21,14 +25,14 @@ class ConversionException extends DBALException
     /**
      * Thrown when a Database to Doctrine Type Conversion fails.
      *
-     * @param string $value
+     * @param mixed  $value
      * @param string $toType
      *
      * @return \Doctrine\DBAL\Types\ConversionException
      */
     public static function conversionFailed($value, $toType)
     {
-        $value = strlen($value) > 32 ? substr($value, 0, 20) . '...' : $value;
+        $value = is_string($value) && strlen($value) > 32 ? substr($value, 0, 20) . '...' : var_export($value, true);
```

Comment:
I'm not sure if we can `var_export()` all the time: this can easily go out of memory if somebody passes a complex object to the type conversion.

Reply:
It's also useless from the end user perspective. Maybe just displaying type/FQN of the value is enough.

Reply:
A test is probably to be added here. Unsure if relevant, but https://3v4l.org/EYdiW could be included to just check that it prints `stdClass`, not much more.

Reply:
I have removed var_export(), leaving only is_string() check. The code is still errorneous as it concatenates mixed with string, but let's address that in other PR.